src/main/java/org/folio/rest/impl/DelayedTasksAPI.java

```diff
@@ -0,0 +1,41 @@
+package org.folio.rest.impl;
+
+import static javax.ws.rs.core.Response.Status;
+
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import org.folio.rest.impl.base.AbstractEmail;
+import org.folio.rest.jaxrs.resource.DelayedTask;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+
+public class DelayedTasksAPI extends AbstractEmail implements DelayedTask {
+
+  public DelayedTasksAPI(Vertx vertx, String tenantId) {
+    super(vertx, tenantId);
+  }
+
+  @Override
+  public void deleteDelayedTaskExpiredMessages(String expirationDate, String status, Map<String, String> headers,
+                                               Handler<AsyncResult<Response>> resultHandler, Context context) {
+    try {
```

Comment:
you can get rid of try/catch staring with succeeded future and call checkExpirationDate in the compose lambda

Reply:
fix